src/lib/util/png.cpp

```diff
@@ -705,7 +705,7 @@ class png_private
 		{
 			// read a chunk
 			std::unique_ptr<std::uint8_t []> chunk_data;
-			std::uint32_t chunk_type, chunk_length;
```

Comment:
The trouble with this change is it now hides the "branch on uninitialised" error from Valgrind or similar if a bug causes `read_chunk` to return `PNGERR_NONE` without setting the `chunk_type` output.  Without initialisation, the `chunk_type` local starts off tainted and won't be marked as set until it's set by `read_chunk`, but now it's always marked as safe, even though it won't behave properly if the zero is actually used for anything.

Reply:
That's true, I didn't see this coming. 
But in case the code would introduce such a bug in the future, I still think it's better to have a deterministic bug than an non-deterministic one. It wouldn't stand out in valgrind, but it would be easily reproducible, which IMO, makes it easier to debug.